### DISCUSSION TITLE
Various Steam Libraries?

### DISCUSSION THREAD

**Discussion by bbuske** [Matoking/protontricks#192] [Q&A]
Hi, 

I am using Manjaro with latest Protontricks. I have Steam installed in the default directory ~/.local/shared/steam but I have a second library on my additional SSD drive. The directory is ~/data/SteamLibrary. When I tried to use Protontricks, I get an error, saying that the directory wasn't found, and it suggests that Protontricks doesn't have permissions.

`BUSB1% protontricks 8170 dotnet40`    

`protontricks (WARNING): Steam library folder /home/bbuske/ext/SteamLibrary not found. Protontricks might not have access to the directory.
`

So, when I try to export $STEAM_DIR, pointing it to the library, it throws an error, complaining, that no valid Steam installation was found in that directory, which makes sense because it's only a library, not a Steam installation.

`BUSB1% STEAM_DIR=/home/bbuske/ext/SteamLibrary/ protontricks 8170 dotnet40 `

`protontricks (ERROR): $STEAM_DIR was provided but didn't point to a valid Steam installation.`

So, how do I get Protontricks, to work with the separate Steam library on the other drive?

**Comment by MatthewForrester** [comment]
Are you running the Flatpak version of Protontricks? If so, then Protontricks may not be able to see the second library, because Flatpak only allows programs to access relevant directories (as a security measure). You need to use Flatseal, which is a tool to control the permissions for Flatpaks. You can [get Flatseal from Flathub](https://flathub.org/apps/com.github.tchx84.Flatseal) (the standard repository for Flatpaks). Open Flatseal, select Protontricks, and give it permission to access the Steam Library on the other drive.

In my case, when I did that, Protontricks seemed to handle multiple libraries automatically.

**Reply by bbuske** [reply]
Hey, I posted that two years ago. And I wasn't using flatpak. 😁 It is no longer an issue though. Thanks anyway.

**Comment by mercster** [comment]
Hey, you posted that three years ago, and it STILL is an issue for me. :P I just installed protontricks manually:

```
merc@quanta:~$ STEAM_DIR=/home/merc/Steam protontricks -l
protontricks (ERROR): $STEAM_DIR was provided but didn't point to a valid Steam installation.
Steam installation directory could not be found.
```

**Reply by cprn** [reply]
On the off chance of it being a trivial issue, doesn't Steam install in something more like `/home/merc/.local/shared/Steam`?

**Reply by mercster** [reply]
> On the off chance of it being a trivial issue, doesn't Steam install in something more like `/home/merc/.local/shared/Steam`?

~~It does by default, but I always make a Steam directory in $HOME and make that my default~~.  ~~Why would I want my games installed somewhere deep into a hidden directory~~?  ~~Never made sense to me~~.  ~~So, yeah~~.  ~~Thanks anyway~~!

Ohhh, you know what?  Maybe it DOES want ~/.steam/steam i.e. Steam's "main" directory, instead of your steam LIBRARY.  Yeah... I guess that makes sense cuz you can have games in multiple libraries.  I'm not sure, but I think you're right.  At any rate, when I posted this I wasa having problems with Steam anyway cuz I was mucking about with `steamcmd`, my protontricks is now working.  Thanks!

**Reply by cprn** [reply]
BTW, it's only mildly related, but for some reason I always just symlinked `~/.local/share/Steam/steamapps/common` to something like `~/SteamGames` instead of using the in-client “move your library” functionality. Maybe it'll prove useful for debugging in the future. I feel like very few people move their library or have more than one, and therefore these mechanics don't get tested enough.

**Reply by mercster** [reply]
> BTW, it's only mildly related, but for some reason I always just symlinked `~/.local/share/Steam/steamapps/common` to something like `~/SteamGames` instead of using the in-client “move your library” functionality. Maybe it'll prove useful for debugging in the future. I feel like very few people move their library or have more than one, and therefore these mechanics don't get tested enough.

Yeah... I've been running Linux for 30+ years and am a retired UNIX systems engineer, so I probably do things differently than a lot of people.  :-)  Cheers.
